Bump cilium-cli to v0.8.4

tklauser opened the pull request.

gandro commented on .github/workflows/conformance-multicluster.yaml:
```diff
@@ -278,11 +278,7 @@ jobs:
-          timeout 5s cilium clustermesh status --context ${{ steps.contexts.outputs.context1 }}
           cilium status --context ${{ steps.contexts.outputs.context2 }}
-          # FIXME: Use of timeout is a workaround for cilium/cilium-cli#384
-          timeout 5s cilium clustermesh status --context ${{ steps.contexts.outputs.context2 }}
```
We still want to dump the status, but thanks to cilium/cilium-cli#384 it won't block forever anymore, so we can just remove the timeout:

```suggestion
          cilium clustermesh status --context ${{ steps.contexts.outputs.context2 }}
```

tklauser replied:
:facepalm: thanks, changed

gandro replied:
Note, this doesn't seem to be fixed on the branch yet.

tklauser replied:
:facepalm: pushed the wrong copy of the branch. Should now be fixed.

nbusseneau reviewed (commented):
Same remark as Sebastian, we need both `cilium clustermesh status` back in, otherwise LGTM :D

nbusseneau commented on .github/workflows/conformance-multicluster.yaml:
```diff
@@ -277,7 +277,7 @@ jobs:
           cilium hubble port-forward --context ${{ steps.contexts.outputs.context1 }}&
           sleep 10s
-          [[ $(pgrep -f "cilium hubble port-forward|kubectl port-forward.*hubble-relay" | wc -l) == 2 ]]
+          [[ $(pgrep -f "cilium hubble port-forward|kubectl port-forward.*hubble-relay" | wc -l) == 4 ]]
```
> `cilium hubble port-forward` is run for both clusters, so we expect to see each process 2 times.

Wait, am I missing something? We don't run `cilium hubble port-forward` for the second cluster, and it doesn't even have Relay 🤔

tklauser replied:
You're completely right, sorry. I must have mixed something up. I'll drop the commit again. Though it's still a mystery to me why that check is consistently failing.

tklauser committed:
.github/workflows: bump cilium-cli to v0.8.4

Signed-off-by: Tobias Klauser <tobias@cilium.io>

tklauser committed:
.github/workflows: remove timeout workaround

This was introduced in commit 2641808d46cc ("ci-multicluster: Fix
post-test information gathering") to work around cilium/cilium-cli#384.
That issue is fixed in the v0.8.4 cilium-cli release, so drop the
workaround.

Suggested-by: Sebastian Wicki <sebastian@isovalent.com>
Signed-off-by: Tobias Klauser <tobias@cilium.io>

nbusseneau reviewed (approved):
:shipit:
